homeassistant/components/maxcube/__init__.py

```diff
@@ -48,40 +50,95 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+PLATFORMS = [
+    Platform.CLIMATE,
+    Platform.BINARY_SENSOR,
+]
+
 
 def setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Establish connection to MAX! Cube."""
-
     if DATA_KEY not in hass.data:
         hass.data[DATA_KEY] = {}
 
-    connection_failed = 0
+    if DOMAIN not in config or CONF_GATEWAYS not in config[DOMAIN]:
+        return True
+
+    _LOGGER.warning(
+        "Configuration of the maxcube platform in YAML is deprecated and will be "
+        "removed in future release; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+
     gateways = config[DOMAIN][CONF_GATEWAYS]
     for gateway in gateways:
-        host = gateway[CONF_HOST]
-        port = gateway[CONF_PORT]
-        scan_interval = gateway[CONF_SCAN_INTERVAL].total_seconds()
-
-        try:
-            cube = MaxCube(host, port, now=now)
-            hass.data[DATA_KEY][host] = MaxCubeHandle(cube, scan_interval)
-        except timeout as ex:
-            _LOGGER.error("Unable to connect to Max!Cube gateway: %s", str(ex))
-            persistent_notification.create(
-                hass,
-                f"Error: {ex}<br />You will need to restart Home Assistant after fixing.",
-                title=NOTIFICATION_TITLE,
-                notification_id=NOTIFICATION_ID,
+        data = {
+            CONF_HOST: gateway[CONF_HOST],
+            CONF_PORT: gateway.get(CONF_PORT, 62910),
+            CONF_SCAN_INTERVAL: gateway.get(
+                CONF_SCAN_INTERVAL, timedelta(seconds=300)
+            ).total_seconds(),
+        }
+
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data=data,
             )
-            connection_failed += 1
+        )
 
-    if connection_failed >= len(gateways):
-        return False
+    return True
 
-    load_platform(hass, Platform.CLIMATE, DOMAIN, {}, config)
-    load_platform(hass, Platform.BINARY_SENSOR, DOMAIN, {}, config)
 
-    return True
+class MaxCubeDeviceUpdater:
```

Comment:
Does this change need to be in the same PR?
It would be much easier to review if this was a preliminary or follow-up PR.